File: python/rating_prediction.py
#!/usr/bin/env ipy

import clr
clr.AddReference("Algorithms.dll")
import MyMediaLite
import MyMediaLite.data      
import MyMediaLite.eval      
import MyMediaLite.io
import MyMediaLite.rating_predictor  

min_rating = 1
max_rating = 5
  
# load the data
user_mapping = MyMediaLite.data.EntityMapping()    
item_mapping = MyMediaLite.data.EntityMapping()
training_data = MyMediaLite.io.RatingPredictionData.Read("u1.base", min_rating, max_rating, user_mapping, item_mapping)
test_data = MyMediaLite.io.RatingPredictionData.Read("u1.test", min_rating, max_rating, user_mapping, item_mapping)     

# set up the recommender
recommender = MyMediaLite.rating_predictor.UserItemBaseline()
recommender.MinRating = min_rating
recommender.MaxRating = max_rating
recommender.Ratings = training_data
recommender.Train()
  
# measure the accuracy on the test data set
print MyMediaLite.eval.RatingEval.EvaluateRated(recommender, test_data)
  
# make a prediction for a certain user and item
print recommender.Predict(user_mapping.ToInternalID(1), item_mapping.ToInternalID(1))

File: website/lib/rating_prediction.rb
#!/usr/bin/env ir

require 'MyMediaLite'

# load the data
train_data = MyMediaLite::IO::RatingData.Read("u1.base")
test_data = MyMediaLite::IO::RatingData.Read("u1.test")

# set up the recommender
recommender = MyMediaLite::RatingPrediction::UserItemBaseline.new()
recommender.Ratings = train_data
recommender.Train()

# measure the accuracy on the test data set
eval_results = MyMediaLite::Eval::Ratings::Evaluate(recommender, test_data)
eval_results.each do |entry|
	puts "#{entry}"
end

# make a prediction for a certain user and item
puts recommender.Predict(1, 1)
